### ISSUE TITLE
BufferedSourceJsonReader.lenient is a private var but doest not expose API to mutate its value

### ISSUE THREAD

**Issue by chao2zhang** [apollographql/apollo-kotlin#4654]
### Version

3.7.4

### Summary

BufferedSourceJsonReader.lenient is a private var but doest not expose API to mutate its value

### Steps to reproduce the behavior

When the server returns a malformed JSON, `Use JsonReader.setLenient(true) to accept malformed JSON` was thrown but the error message didn't indicate a fix in practise

### Logs

```
(Your logs here)
```

**Comment by BoD** [comment]
Hi!

Indeed, the message is misleading since there is no way to do that!
On the other hand I'm not sure it would be wise to add the option as it would be preferable for the backend to emit valid JSON in all cases. I'd rather replace the error message by "Invalid JSON" or something. For the sake of curiosity, could you share which part of the JSON is triggering the error?

**Comment by chao2zhang** [comment]
In my synthetic test cases, the following can trigger `Use JsonReader.setLenient(true) to accept malformed JSON` exceptions

```
            {
              "errors": [;]
              "data": null
            }
```

**Comment by BoD** [comment]
I see, thanks! The semicolon and missing comma. 

To help us better understand, does this also happen with actual payloads coming from the backend or is it limited to tests?

In any case I think it will make more sense to not have a lenient mode at all in the parser.

**Comment by chao2zhang** [comment]
It happened with the actual payloads, but we need to wait for a week for our app to be released so that we can capture the actual payload.

**Comment by BoD** [comment]
Thanks for clarifying! Please update the issue if/when you have access to the payload, just so we're sure that there isn't an actual issue with the parser.

**Comment by chao2zhang** [comment]
From our logging, we either have
- Body as an empty string
- Body length exceeding 16000KB which is forbidden to be fully logged in our observability tools
  - The query name aligns with what's reported https://github.com/apollographql/apollo-kotlin/issues/4570
  - Our guess is that in those cases, `error` and `data` are both present, and `data` section is too large to be reported.
- The only useful stacktrace for the non-empty Body I see is, where I think is the stream is closed before fully parsing
```
Caused by: okhttp3.internal.http2.StreamResetException: stream was reset: CANCEL
	at okhttp3.internal.http2.Http2Stream$FramingSource.read(Http2Stream.kt:54)
	at okhttp3.internal.connection.Exchange$ResponseBodySource.read(Exchange.kt:14)
	at okio.RealBufferedSource.read(RealBufferedSource.kt:4)
	at okio.RealBufferedSource.exhausted(RealBufferedSource.kt:21)
	at okio.InflaterSource.readOrInflate(InflaterSource.kt:51)
	at okio.InflaterSource.read(InflaterSource.kt:6)
	at okio.GzipSource.read(GzipSource.kt:321)
	at okio.RealBufferedSource.indexOfElement(RealBufferedSource.kt:35)
	at com.apollographql.apollo3.api.json.BufferedSourceJsonReader.nextQuotedValue(BufferedSourceJsonReader.kt:4)
	at com.apollographql.apollo3.api.json.BufferedSourceJsonReader.nextString(BufferedSourceJsonReader.kt:94)
	at com.apollographql.apollo3.api.json.-JsonReaders.readAny(JsonReaders.kt:101)
	at com.apollographql.apollo3.api.json.-JsonReaders.readAny(JsonReaders.kt:89)
	at com.apollographql.apollo3.api.json.-JsonReaders.readAny(JsonReaders.kt:89)
	at com.apollographql.apollo3.api.ObjectAdapter.fromJson(Adapters.kt:39)
	at com.apollographql.apollo3.api.ListAdapter.fromJson(Adapters.kt:5)
	at com.apollographql.apollo3.api.ListAdapter.fromJson(Adapters.kt:1)
	at com.apollographql.apollo3.api.NullableAdapter.fromJson(Adapters.kt:26)
	at 
```

**Comment by BoD** [comment]
Thanks a lot for reporting back!

It looks like they're all network errors, broadly speaking - that explains why the JSON parsing could fail.

In #4656 I removed the misleading lenient messages, I will merge it then.

**Comment by BoD** [comment]
#4556 has been released as of [v3.7.5](https://github.com/apollographql/apollo-kotlin/releases/tag/v3.7.5) - the misleading error messages have been removed.
